### PR TITLE
Job actions ( Schedule Interview, Send Offer And Regret Letter )

### PR DESCRIPTION
**Pull request by rajtejani**

### CODE REVIEW

File: helpers/constants.js [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+  expertiseLevel: {
+    FRESHER: "FRESHER",
+    INTERMIDIATE: "INTERMIDIATE",
+    EXPERIENCED: "EXPERIENCED",
+  },
+  applicationStatus: {
+    APPLIED: "APPLIED",
+    INTERVIEWING: "INTERVIEWING",
+    HIRED: "HIRED",
+    REJECTED: "REJECTED",
```
Comment (n4alphapesto) [inline helpers/constants.js:26]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		REJECTED: "REJECTED",
```

File: models/JobModel.js [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
 
+const applicantSchema = new Schema({
+  candidate: {
+    type: String,
+    required: true,
+  },
+  status: {
+    type: String,
+    enum: [constants.applicationStatus],
+    default: constants.applicationStatus.APPLIED,
```
Comment (n4alphapesto) [inline models/JobModel.js:15]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		default: constants.applicationStatus.APPLIED,
```

File: helpers/constants.js [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
+  confirmEmails: {
+    from: "a0k03lf@gmail.com",
+  },
+  userTypes: {
+    RECRUITER: "RECRUITER",
+    JOBSEEKER: "JOBSEEKER",
+  },
+  experienceType: {
+    FRESHER: "FRESHER",
+    PROFESSIONAL: "PROFESSIONAL",
```
Comment (n4alphapesto) [inline helpers/constants.js:15]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		PROFESSIONAL: "PROFESSIONAL",
```

File: controllers/JobController.js [hunk 4]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
...
+      UserModel.findOne({ _id: candidateId }, (error, user) => {
+        if (error) {
+          return apiResponse.ErrorResponse(res, "Operation Failed.", error);
+        }
+
+        if (!user) {
+          return apiResponse.ErrorResponse(res, "Candidate doesn't exists.");
+        }
+
+        JobModel.updateOne(
```
Comment (n4alphapesto) [inline controllers/JobController.js:313]: Codacy has a fix for the issue: [Expected indentation of 6 tabs but found 12 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
						JobModel.updateOne(
```

File: models/JobModel.js [hunk 5]
```diff
@@ -13,6 +26,10 @@ const JobSchema = new Schema(
     locations: [String],
     vacancies: { type: Number, require: true },
     isDeleted: { type: Boolean, default: false },
+    applicants: {
+      type: [applicantSchema],
```
Comment (n4alphapesto) [inline models/JobModel.js:30]: Codacy has a fix for the issue: [Expected indentation of 3 tabs but found 6 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
			type: [applicantSchema],
```

File: helpers/constants.js [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
-		INTERVIEWING: "INTERVIEWING",
-		HIRING: "HIRING",
-		REJECTED: "REJECTED"
-	}
+  admin: {
+    name: "admin",
+    email: "a0k03lf@gmail.com",
+  },
+  confirmEmails: {
+    from: "a0k03lf@gmail.com",
```
Comment (n4alphapesto) [inline helpers/constants.js:7]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		from: "a0k03lf@gmail.com",
```

File: controllers/JobController.js [hunk 7]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
...
+        if (!user) {
+          return apiResponse.ErrorResponse(res, "Candidate doesn't exists.");
+        }
+
+        console.log(" jobid & candidateId ", {
+          jobId: mongoose.Types.ObjectId(jobId),
+          candidateId,
+        });
+        JobModel.findOneAndUpdate(
+          {
```
Comment (n4alphapesto) [inline controllers/JobController.js:221]: Codacy has a fix for the issue: [Expected indentation of 4 tabs but found 8 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
				{
```

File: controllers/JobController.js [hunk 8]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
...
+                message
+              )
+              .then(() => {
+                return apiResponse.successResponse(res, "Interview scheduled.");
+              })
+              .catch((error) => {
+                if (error) {
+                  return apiResponse.ErrorResponse(
+                    res,
+                    "Operation Failed.",
```
Comment (n4alphapesto) [inline controllers/JobController.js:259]: Codacy has a fix for the issue: [Expected indentation of 10 tabs but found 20 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
										"Operation Failed.",
```

File: models/JobModel.js [hunk 9]
```diff
@@ -13,6 +26,10 @@ const JobSchema = new Schema(
     locations: [String],
     vacancies: { type: Number, require: true },
     isDeleted: { type: Boolean, default: false },
+    applicants: {
```
Comment (n4alphapesto) [inline models/JobModel.js:29]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		applicants: {
```

File: models/JobModel.js [hunk 10]
```diff
@@ -1,9 +1,22 @@
...
 const SkillSchema = require("./SkillModel");
 
+const applicantSchema = new Schema({
+  candidate: {
+    type: String,
+    required: true,
+  },
+  status: {
+    type: String,
+    enum: [constants.applicationStatus],
```
Comment (n4alphapesto) [inline models/JobModel.js:14]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		enum: [constants.applicationStatus],
```

File: models/JobModel.js [hunk 11]
```diff
@@ -1,9 +1,22 @@
...
 const Schema = mongoose.Schema;
 const DataTypes = Schema.Types;
 const SkillSchema = require("./SkillModel");
 
+const applicantSchema = new Schema({
+  candidate: {
+    type: String,
+    required: true,
+  },
+  status: {
```
Comment (n4alphapesto) [inline models/JobModel.js:12]: Codacy has a fix for the issue: [Expected indentation of 1 tab but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
	status:   {
```

File: helpers/constants.js [hunk 12]
```diff
@@ -1,28 +1,28 @@
...
+  },
+  userTypes: {
+    RECRUITER: "RECRUITER",
+    JOBSEEKER: "JOBSEEKER",
```
Comment (n4alphapesto) [inline helpers/constants.js:11]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		JOBSEEKER: "JOBSEEKER",
```

File: helpers/constants.js [hunk 13]
```diff
@@ -1,28 +1,28 @@
...
+  },
+  expertiseLevel: {
+    FRESHER: "FRESHER",
+    INTERMIDIATE: "INTERMIDIATE",
+    EXPERIENCED: "EXPERIENCED",
+  },
+  applicationStatus: {
+    APPLIED: "APPLIED",
+    INTERVIEWING: "INTERVIEWING",
+    HIRED: "HIRED",
```
Comment (n4alphapesto) [inline helpers/constants.js:25]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		HIRED: "HIRED",
```

File: helpers/constants.js [hunk 14]
```diff
@@ -1,28 +1,28 @@
...
-		EXPERIENCED: "EXPERIENCED"
-	},
-	applicationStatus: {
-		APPLIED: "APPLIED",
-		INTERVIEWING: "INTERVIEWING",
-		HIRING: "HIRING",
-		REJECTED: "REJECTED"
-	}
+  admin: {
+    name: "admin",
```
Comment (n4alphapesto) [inline helpers/constants.js:3]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		name: "admin",
```

File: helpers/constants.js [hunk 15]
```diff
@@ -1,28 +1,28 @@
...
+  experienceType: {
+    FRESHER: "FRESHER",
+    PROFESSIONAL: "PROFESSIONAL",
+  },
+  expertiseLevel: {
+    FRESHER: "FRESHER",
+    INTERMIDIATE: "INTERMIDIATE",
+    EXPERIENCED: "EXPERIENCED",
+  },
+  applicationStatus: {
```
Comment (n4alphapesto) [inline helpers/constants.js:22]: Codacy has a fix for the issue: [Expected indentation of 1 tab but found 2 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
	applicationStatus: {
```

File: helpers/constants.js [hunk 16]
```diff
@@ -1,28 +1,28 @@
...
-		APPLIED: "APPLIED",
-		INTERVIEWING: "INTERVIEWING",
-		HIRING: "HIRING",
-		REJECTED: "REJECTED"
-	}
+  admin: {
+    name: "admin",
+    email: "a0k03lf@gmail.com",
+  },
+  confirmEmails: {
```
Comment (n4alphapesto) [inline helpers/constants.js:6]: Codacy has a fix for the issue: [Expected indentation of 1 tab but found 2 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
	confirmEmails: {
```

File: models/JobModel.js [hunk 17]
```diff
@@ -1,9 +1,22 @@
 const mongoose = require("mongoose");
-
+const { constants } = require("../helpers/constants");
 const Schema = mongoose.Schema;
 const DataTypes = Schema.Types;
 const SkillSchema = require("./SkillModel");
 
+const applicantSchema = new Schema({
+  candidate: {
```
Comment (n4alphapesto) [inline models/JobModel.js:8]: Codacy has a fix for the issue: [Expected indentation of 1 tab but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
	candidate: {
```

File: controllers/JobController.js [hunk 18]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
...
+                    error
```
Comment (n4alphapesto) [inline controllers/JobController.js:260]: Codacy has a fix for the issue: [Expected indentation of 10 tabs but found 20 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
										error
```

File: controllers/JobController.js [hunk 19]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
     }
   },
 ];
+
+// This function sends interview scheduling mail to cadidate
+exports.scheduleInterview = [
+  auth,
+  body("jobId", "jobId must not be empty.").notEmpty(),
```
Comment (n4alphapesto) [inline controllers/JobController.js:174]: Codacy has a fix for the issue: [Expected indentation of 1 tab but found 2 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
	body("jobId", "jobId must not be empty.").notEmpty(),
```

File: controllers/JobController.js [hunk 20]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
...
+            "applicants.candidate": candidateId,
+            "applicants.status": constants.applicationStatus.APPLIED,
+          },
+          {
+            $set: {
+              "applicants.$.status": constants.applicationStatus.INTERVIEWING,
+            },
+          },
+          { new: true },
+          (error, result) => {
```
Comment (n4alphapesto) [inline controllers/JobController.js:232]: Codacy has a fix for the issue: [Expected indentation of 4 tabs but found 8 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
				(error, result) => {
```

File: models/JobModel.js [hunk 21]
```diff
@@ -1,9 +1,22 @@
...
+    type: String,
+    required: true,
```
Comment (n4alphapesto) [inline models/JobModel.js:10]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 8 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		required: true,
```

File: helpers/constants.js [hunk 22]
```diff
@@ -1,28 +1,28 @@
...
-		INTERMIDIATE: "INTERMIDIATE",
-		EXPERIENCED: "EXPERIENCED"
-	},
-	applicationStatus: {
-		APPLIED: "APPLIED",
-		INTERVIEWING: "INTERVIEWING",
-		HIRING: "HIRING",
-		REJECTED: "REJECTED"
-	}
+  admin: {
```
Comment (n4alphapesto) [inline helpers/constants.js:2]: Codacy has a fix for the issue: [Expected indentation of 1 tab but found 2 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
	admin: {
```

File: helpers/constants.js [hunk 23]
```diff
@@ -1,28 +1,28 @@
...
+    FRESHER: "FRESHER",
+    PROFESSIONAL: "PROFESSIONAL",
+  },
+  expertiseLevel: {
+    FRESHER: "FRESHER",
+    INTERMIDIATE: "INTERMIDIATE",
+    EXPERIENCED: "EXPERIENCED",
+  },
+  applicationStatus: {
+    APPLIED: "APPLIED",
```
Comment (n4alphapesto) [inline helpers/constants.js:23]: Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 4 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
		APPLIED: "APPLIED",
```

File: controllers/JobController.js [hunk 24]
```diff
@@ -164,3 +167,289 @@ exports.addJob = [
...
+            _id: mongoose.Types.ObjectId(jobId),
+            "applicants.candidate": candidateId,
+            "applicants.status": constants.applicationStatus.APPLIED,
+          },
+          {
+            $set: {
+              "applicants.$.status": constants.applicationStatus.INTERVIEWING,
+            },
+          },
+          { new: true },
```
Comment (n4alphapesto) [inline controllers/JobController.js:231]: Codacy has a fix for the issue: [Expected indentation of 7 tabs but found 14 spaces.](https://app.codacy.com/gh/n4alphapesto/hirebook-backend/pullRequest?prid=7787263)
```suggestion
							{ new: true },
```

**Review by avijsdev — APPROVED** [review]

**Pull request merged**